brightener.cpp

```diff
@@ -31,15 +31,16 @@ shared_ptr<Image> BrightenWholeImage(shared_ptr<Image> inputImage, int& attenuat
     return brightenedImage;
 }
 
-shared_ptr<Image> AddBrighteningImage(shared_ptr<Image> inputImage, shared_ptr<Image> imageToAdd,
+shared_ptr<BrightenedImage> AddBrighteningImage(const shared_ptr<RawImage> inputImage,const shared_ptr<Image> imageToAdd,
     int& attenuatedPixelCount) {
-    // Try converting this into an exception, so callers don't always need to check the returned bool
-    // if (imageToAdd->m_rows != m_inputImage->m_rows || imageToAdd->m_columns != m_inputImage->m_columns) {
-    //     return false;
-    // }
+
+     if (imageToAdd->m_rows != inputImage->m_rows || imageToAdd->m_columns != inputImage->m_columns) {
+         throw "Image dimensions does not match";
+     }
+
     attenuatedPixelCount = 0;
     auto brightenedImage =
-        make_shared<Image>(inputImage->m_rows, inputImage->m_columns,
+        make_shared<BrightenedImage>(inputImage->m_rows, inputImage->m_columns,
```

Comment:
You probably have a duplication here? Try this: copy the code in this file into ChatGPT and ask it to remove duplication. The code it generates would probably have compilation issues. Tell it those issues and ask it to fix. Let's see where that takes us. BTW this comment sounds like fun, but actually I am serious.